servicex_app/servicex/models.py

```diff
@@ -215,6 +218,24 @@ def lookup(cls, key: Union[str, int]) -> Optional['TransformRequest']:
         except NoResultFound:
             return None
 
+    @classmethod
+    def file_transformed_successfully(cls, key: Union[str, int]) -> None:
+        req = cls.query.filter_by(request_id=key).one()
+        req.files_completed = cls.files_completed + 1
+        db.session.commit()
+
+    @classmethod
+    def file_transformed_unsuccessfully(cls, key: Union[str, int]) -> None:
+        req = cls.query.filter_by(request_id=key).one()
+        req.files_failed = cls.files_failed + 1
+        db.session.commit()
+
+    @classmethod
+    def add_a_file(cls, key) -> None:
```

Comment:
Do you know what happens if another process updates this record while we are doing this? I assume the commit fails. How do we make sure we don't lose updates?

Reply:
commit always does flush first, so we never lose updates.